### PR TITLE
LineBuffer API change (re-take)

### PR DESCRIPTION
**Pull request by MarcinOrlowski**
This PR removes access to LineBuffer's default constructor, enforcing use of getBuffer() and getHdlBuffer() static calls.

This is #1113 done again using fresh codebase as PR fork diverged too much and it is easier and cheaper to redo the PR.

* #1113

### CODE REVIEW

**Comment by BFH-ktt1** [comment]
@MarcinOrlowski : I would propose still to wait until #1114 has landed, as there are again a lot of changes in the HDL-generators. After I finish #1114 I can go through all the HDL-generators (in this PR) to clean-up. I would propose that you only make the changes to Linebuffer in this PR to prevent a lot of "double/tripple" work

**Comment by MarcinOrlowski** [comment]
> I would propose that you only make the changes to Linebuffer in this PR to prevent a lot of "double/tripple" work

Done.

**Comment by MarcinOrlowski** [comment]
@BFH-ktt1 can you please merge what qualifies?

**Review by BFH-ktt1 — APPROVED** [review]

**Pull request merged**
